### ISSUE TITLE
Non-empty sttp.apispec.asyncapi.Message.examples or MessageTrait.examples yields invalid specification

### ISSUE THREAD

**Issue by zorba128** [softwaremill/sttp-apispec#174]
Adding example to either `Message` or `MessageTrait` produces invalid specification.

Generated specs:
```
 messages:
    WsRequest:
      payload:
        $ref: '#/components/schemas/WsRequest'
      contentType: application/json
      examples:
      - payload:
        - wsTag: 1
          channel: notifications
          requestType: Subscribe
      - payload:
        - wsTag: 2
          channel: notifications
          requestType: Unsubscribe
```

And should be:
```
 messages:
    WsRequest:
      payload:
        $ref: '#/components/schemas/WsRequest'
      contentType: application/json
      examples:
      - payload:
          wsTag: 1
          channel: notifications
          requestType: Subscribe
      - payload:
          wsTag: 2
          channel: notifications
          requestType: Unsubscribe
```

Original,broken definition is following:
```
case class SingleMessage(..., examples: List[Map[String, List[ExampleValue]]] = Nil, ...) extends Message
```

It should be either simplified to:
```
case class SingleMessage(..., examples: List[ListMap[String, ExampleValue]] = Nil, ...) extends Message
```

or simply to follow specification [asyncapi spec](https://v2.asyncapi.com/docs/reference/specification/v2.6.0#messageExampleObject), MessageExampleObject section. This would also make example name/description support better (now only `payload` is used).

```
// TODO: headers, extensions
case class MessageExample(payload: ExampleValue, name: Option[String], summary: Option[String])
case class SingleMessage(..., examples: List[MessageExample], ...) extends Message
```

See PR https://github.com/softwaremill/sttp-apispec/pull/175

**Comment by adamw** [comment]
Thanks for working on this :)

**Comment by zorba128** [comment]
Actually this one, plus some minor updates on tapir side would fix issue with examples - but I have no idea how to proceed as both will be breaking changes.

**Comment by adamw** [comment]
It's a 0.x version, so breaking changes are fine, mima is there so that we know when to bump the minor version (vs. patch)

**Comment by zorba128** [comment]
It seems PR works fine - generated spec worked fine in AsyncAPI Studio:
![image](https://github.com/softwaremill/sttp-apispec/assets/6301362/4e89a48b-2903-4f4b-9834-2454590c5c87)

**Comment by adamw** [comment]
Released in 0.11.0
